### PR TITLE
Migrate txns to supabase

### PR DESCRIPTION
**Pull request by sipec**
need to test

Resolves MAN-1414

### CODE REVIEW

File: backend/shared/src/txn/run-txn.ts [hunk 1]
```diff
@@ -1,149 +1,195 @@
...
+        txn.amount,
+        txn.fromId,
+        txn.toId,
+        txn.fromType,
+        txn.toType,
+        txn.category,
+      ]
+    )
+    .catch((e) => {
+      logFailedToRecordTxn(txn)
```
Comment (jahooma) [inline backend/shared/src/txn/run-txn.ts:158]: From discussion: Would be good to trigger these in dev and search the logs for them

File: backend/shared/src/onboarding-helpers.ts [hunk 2]
```diff
@@ -105,17 +108,22 @@ const sendNextDayManaBonus = async (
...
     }
-    const manaBonusTxn = await runTxnFromBank(transaction, signupBonusTxn)
-    if (manaBonusTxn.status != 'error' && manaBonusTxn.txn) {
-      transaction.update(toDoc, {
-        manaBonusSent: true,
-        weeklyTrendingEmailSent: true, // not yet, but about to!
+
+    const manaBonusTxn = await pg
+      .tx((tx) => {
+        return runTxnFromBank(tx, signupBonusTxn)
```
Comment (jahooma) [inline backend/shared/src/onboarding-helpers.ts:114]: Seems like this nests the firestore transactions. That would break?

Comment (jahooma) [reply backend/shared/src/onboarding-helpers.ts:114]: Ok, this does a firestore write not a txn.

Comment (jahooma) [reply backend/shared/src/onboarding-helpers.ts:114]: [c] Should test creating a user to make sure this works [J] Yes

File: backend/api/src/boost-market.ts [hunk 3]
```diff
@@ -41,57 +39,39 @@ export const boostmarket = authEndpoint(async (req, auth) => {
...
-  } catch (e) {
-    log('error adding txn! reversing funds to ad.')
 
-    await pg.none(
-      `update market_ads
-      set funds = market_ads.funds - $1
-      where id = $2`,
-      [totalCost, id]
-    )
-    log(`done subtracting ${totalCost} from ad ${id}`)
```
[the remark this answers is not in the document]
Comment (jahooma) [reply backend/api/src/boost-market.ts:90]: [c] this is to put it back if there was an error. But now it's in a tx

**Review by jahooma — APPROVED** [review]
Looks good! Thanks Sinclair.

Monitoring for errors is very important after this.

Would be good to keep tabs on any errors. We should have a plan for what exactly to query for.
[c] Could create a dashboard from these logs.
- If it fails in firebase transaction, will roll back correctly
- If a supabase part fails, it will throw an error
- If firestore succeeds and supabase fails, we do log if the txn insert failed. [c] Can check if there are other postgres things after firestore

One idea: Wrap pg.tx in our runTxn code. Allow the caller to pass a pre-query (function that takes tx) that runs before txn is created.

[c] This code might not be in this state for that long. Once we migrate users then balance can be in the pg.tx transaction.

File: backend/shared/src/onboarding-helpers.ts (hunk 2 above)
Comment (sipec) [reply backend/shared/src/onboarding-helpers.ts]: it did break but now it's all un-nested

**Comment by sipec** [comment]
TXN Migration plan
- Merge and wait for frontend to deploy
- Alter table id default, created_time default not null, fs_updated time drop not null
- Recreate txn_populate_cols (don’t populate created time)
- Deploy API
- Redeploy logCollections, onUpdatePrivateUser
- Wait for firebase writes to finish
- Alter table add not nulls for other columns

**Pull request merged**